openedx/core/djangoapps/auth_exchange/views.py

```diff
@@ -145,7 +146,14 @@ def post(self, request):
         if not hasattr(request.user, 'backend'):
             request.user.backend = self._get_path_of_arbitrary_backend_for_user(request.user)
 
-        if not self._is_grant_password(request.auth):
+        access_token = request.auth
+        is_jwt_authenticated = isinstance(request.successful_authenticator, JwtAuthentication)
+        if is_jwt_authenticated:
+            refresh_token = dot_models.RefreshToken.objects.filter(token=request.data.get('refresh_token')).first()
```

Comment:
Can you explain why and how you think this refresh_token necessarily relates to the current JWT token? Or enough to do the later check? Does it matter if it isn't the right one? Seems like it would.

Reply:
Better yet, is the access token associated to the refresh token in the DB unused in the case of a JWT access token request? Can we have DOT instead store the real JWT access token? If it did, could this code work unchanged?

Reply:
Searches lead me to: https://github.com/jazzband/django-oauth-toolkit/pull/467

Reply:
According to the current user flow this refresh token relates to the access_token. We can have a discussion on this.

Reply:
* I think it will be safer and cleaner to add the grant type to the JWT itself and be able to check it directly. 
* I updated the [Mobile JWT Confluence doc](https://openedx.atlassian.net/wiki/spaces/AC/pages/3351609433/Mobile+authentication+with+JWTs) with this detail, and other notes about changes that are required.
* I added a checklist to the PR description of updates we want to make.